IRescue/UserLocalisation/Feedback/PositionMotionFeedbackProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// <copyright file="PositionMotionFeedbackProvider.cs" company="Delft University of Technology">
+// Copyright (c) Delft University of Technology. All rights reserved.
+// </copyright>
+namespace IRescue.UserLocalisation.Feedback
+{
+    using System;
+    using System.Collections.Generic;
+
+    using IRescue.Core.DataTypes;
+
+    /// <summary>
+    /// Provides feedback about the positional motion of the user.
+    /// </summary>
+    public class PositionMotionFeedbackProvider : IPositionFeedbackReceiver, IVelocityFeedbackProvider
+    {
+        /// <summary>
+        /// List with all registered <see cref="IVelocityFeedbackReceiver"/>s.
+        /// </summary>
+        private readonly List<IVelocityFeedbackReceiver> velReceivers;
+
+        /// <summary>
+        /// The previous received position feedback.
+        /// </summary>
+        private FeedbackData<Vector3>? previousData;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PositionMotionFeedbackProvider"/> class.
+        /// </summary>
+        public PositionMotionFeedbackProvider()
+        {
+            this.velReceivers = new List<IVelocityFeedbackReceiver>();
+        }
+
+        /// <inheritdoc/>
+        public void NotifyPositionFeedback(FeedbackData<Vector3> data)
+        {
+            if ((this.previousData != null) && (data.TimeStamp <= this.previousData.Value.TimeStamp))
+            {
+                return;
+            }
+
+            FeedbackData<Vector3>? velocityFeedback = this.CalculateVelocityFeedback(data);
+
+            if (velocityFeedback != null)
+            {
+                for (int index = 0; index < this.velReceivers.Count; index++)
+                {
+                    IVelocityFeedbackReceiver receiver = this.velReceivers[index];
+                    receiver.NotifyVelocityFeedback(velocityFeedback.Value);
+                }
+            }
+
+            this.previousData = data;
+        }
+
+        /// <summary>
+        /// Registers a feedback receiver so it will be notified when new feedback is available.
+        /// </summary>
+        /// <param name="receiver">The receiver to register.</param>
+        public void RegisterReceiver(IVelocityFeedbackReceiver receiver)
+        {
+            if (this.velReceivers.Contains(receiver))
+            {
+                return;
+            }
+
+            this.velReceivers.Add(receiver);
+        }
```

Comment:
What about

if (!this.velReceivers.Contains(receiver))
            {
             this.velReceivers.Add(receiver);
            }